_posts/2019/2019-12-16-how-to-enhance-an-application-using-firebase-auth-and-firebase-functions.md

```diff
@@ -0,0 +1,202 @@
+---
+layout: post
+title: How to enhance an application using Firebase Auth and Firebase Functions
+date: 2019-12-16
+header_image: public/firebase-post-2.jpg
+header_position: top
+header_overlay: true
+category: coding
+tags: ["firebase", "app", "cloud functions", "app development"]
+authors: ["Florian O."]
+about_authors: ["foellerich"]
+---
+
+In the [last part of this series](/blog/coding/how-to-bootstrap-an-application-with-authentication-using-firebase/){:target="_blank"}, we went through all the steps required to bootstrap an application with authentication.
+Our intermediate result was a login UI that can be used to distinguish users.
+Furthermore, we were able to give API access only to registered users.
+In this part, we'll take care of a database creation and security improvements.
+
+## Step 1 - Database creation
+
+Via the authentication service Firebase Auth, we receice tokens for different shops.
+Only the specific merchant of the shop should have access to this token.
+To do so, we first need to create a database in the Firebase console (Develop > Database).
+Then, we create a Cloud Firestore database.
+For fast results, we choose the test mode for now.
+We'll secure the data in a later step.
```

Comment:
I just recognized that I never wrote about this step :/

Reply:
Mmh... Made a proposal how to handle this... Or would you vote for not mentioning it at all? :D